src/test/java/com/google/sps/CommentRetrievalTest.java

```diff
@@ -0,0 +1,149 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.mock;
+
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.model.CommentThread;
+import com.google.api.services.youtube.model.CommentThreadListResponse;
+import com.google.sps.servlets.utils.YouTubeCommentRetriever;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+// Using wildcard import since documentation is unclear which package
+// "USES_DEEP_STUBS" is imported from
+
+@RunWith(JUnit4.class)
+public class CommentRetrievalTest {
+
+  private final int HUNDRED = 100;
+  private final int EXCESS_HUNDRED = 200;
+  private final int MAX_COMMENTS_PER_TOKEN = 100;
+  private YouTubeCommentRetriever commentRetriever;
+  // These url's no longer do anything; rather, serve as parameter for the call
+  private final String POPULAR_VIDEO_URL = "dQw4w9WgXcQ";
+  private final String UNPOPULAR_VIDEO_URL = "cA-arJ0T6L4";
+  private final String NEXT_PAGE_TOKEN = "Some Page Token";
+  private final String FIRST_TOKEN_COMMENT = "First";
+  private final String SECOND_TOKEN_COMMENT = "Second";
+
+  public void setUp(int numExpectedComments) throws Exception {
+    setUp(numExpectedComments, null);
+  }
+  // Simulate a CommentThreadListResponse with exactly as many comments as expected
+  public void setUp(int numExpectedComments, String nextPageToken) throws Exception {
+    YouTube mockedYoutube = mock(YouTube.class, RETURNS_DEEP_STUBS);
+    YouTube.CommentThreads.List mockedCommentThreadList =
+        mock(YouTube.CommentThreads.List.class, RETURNS_DEEP_STUBS);
+    // This is only enough for two calls, all these tests require
+    // Return on the first call either 100 comments or the desired number.
+    // If there's no next page token, return nothing on the second call,
+    // Otherwise return the remainder of comments
+    CommentThreadListResponse firstResponse =
+        mockThreadListResponse(
+            Math.min(numExpectedComments, MAX_COMMENTS_PER_TOKEN),
+            nextPageToken,
+            FIRST_TOKEN_COMMENT);
+    CommentThreadListResponse secondResponse =
+        mockThreadListResponse(
+            (nextPageToken == null)
+                ? 0
+                : Math.min(numExpectedComments - MAX_COMMENTS_PER_TOKEN, MAX_COMMENTS_PER_TOKEN),
+            nextPageToken,
+            SECOND_TOKEN_COMMENT);
+    when(mockedCommentThreadList.execute()).thenReturn(firstResponse, secondResponse);
```

Comment:
I think you can make this block a little bit simpler if you remove the `nextPageToken` argument and instead have your mock only return a `secondResponse` if `numExpectedComments` is greater than 100.

Any anyways since the `nextPageToken` is always either `NEXT_PAGE_TOKEN` or `null` having it be changeable does not offer that much value.